src/main/java/uk/ac/ebi/eva/commons/core/models/VariantStats.java

```diff
@@ -0,0 +1,714 @@
+    }
+
+    public float getCasesPercentRecessive() {
+        return casesPercentRecessive;
+    }
+
+    void setCasesPercentRecessive(float casesPercentRecessive) {
+        this.casesPercentRecessive = casesPercentRecessive;
+    }
+
+    public float getControlsPercentRecessive() {
+        return controlsPercentRecessive;
+    }
+
+    void setControlsPercentRecessive(float controlsPercentRecessive) {
+        this.controlsPercentRecessive = controlsPercentRecessive;
+    }
+
+    public boolean isTransition() {
+        switch (refAllele.toUpperCase()) {
+            case "C":
+                return altAllele.equalsIgnoreCase("T");
+            case "T":
+                return altAllele.equalsIgnoreCase("C");
+            case "A":
+                return altAllele.equalsIgnoreCase("G");
+            case "G":
+                return altAllele.equalsIgnoreCase("A");
+            default:
+                return false;
+        }
+    }
+
+    public boolean isTransversion() {
+        switch (refAllele.toUpperCase()) {
+            case "C":
+                return !altAllele.equalsIgnoreCase("T");
+            case "T":
+                return !altAllele.equalsIgnoreCase("C");
+            case "A":
+                return !altAllele.equalsIgnoreCase("G");
+            case "G":
+                return !altAllele.equalsIgnoreCase("A");
+            default:
+                return false;
+        }
+    }
+
+    public VariantHardyWeinbergStats getHw() {
+        return hw;
+    }
+
+    public boolean hasPassedFilters() {
+        return passedFilters;
+    }
+
+    void setPassedFilters(boolean passedFilters) {
+        this.passedFilters = passedFilters;
+    }
+
+    public float getQuality() {
+        return quality;
+    }
+
+    void setQuality(float quality) {
+        this.quality = quality;
+    }
+
+    public int getNumSamples() {
+        return numSamples;
+    }
+
+    void setNumSamples(int numSamples) {
+        this.numSamples = numSamples;
+    }
+
+    @Override
+    public String toString() {
+        return "VariantStats{"
+                + "refAllele='" + refAllele + '\''
+                + ", altAllele='" + altAllele + '\''
+                + ", mafAllele='" + mafAllele + '\''
+                + ", mgfAllele='" + mgfGenotype + '\''
+                + ", maf=" + maf
+                + ", mgf=" + mgf
+                + ", missingAlleles=" + missingAlleles
+                + ", missingGenotypes=" + missingGenotypes
+                + ", mendelinanErrors=" + mendelianErrors
+                + ", casesPercentDominant=" + casesPercentDominant
+                + ", controlsPercentDominant=" + controlsPercentDominant
+                + ", casesPercentRecessive=" + casesPercentRecessive
+                + ", controlsPercentRecessive=" + controlsPercentRecessive
+                + '}';
+    }
+
+    public VariantStats calculate(List<Map<String, String>> samplesData, Map<String, String> attributes,
```

Comment:
Could this be documented? I know it's calculating the actual variant stats and returns itself, but it's 90 lines long and the details of what it's doing aren't immediately clear without reading through the whole method.